hudi-client/hudi-client-common/src/main/java/org/apache/hudi/keygen/BaseKeyGenerator.java

```diff
@@ -71,11 +84,25 @@ public final HoodieKey getKey(GenericRecord record) {
     }).collect(Collectors.toList());
   }
 
+  public List<String> getIndexKeyFields() {
+    return indexKeyFields;
+  }
+
   public List<String> getRecordKeyFields() {
     return recordKeyFields;
   }
 
   public List<String> getPartitionPathFields() {
     return partitionPathFields;
   }
+
+  protected void validateIndexKeyField() {
+    if (indexKeyFields.size() != 0) {
```

Comment:
would indexKeyFields be empty or mandatory when using hash index?

Reply:
Mandatory. If empty, will throw a hoodieException with related info.

Reply:
so here if indexKeyFields is null or empty, should throw exception here?

Reply:
and please add uts to cover the changes.

Reply:
Check the bucket index key field config when building the HoodieIndexConfig and add the relevant uts in `TestHoodieIndex`